plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -247,49 +248,81 @@ public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStati
-                                    .withDatabaseName(partition.getDatabaseName())
-                                    .withTableName(partition.getTableName())
-                                    .withPartitionValues(partition.getValues())
-                                    .withColumnStatisticsList(columnChunk)), writeExecutor))
-                    .collect(toUnmodifiableList());
+        columnChunks.forEach(columnChunk ->
+                operationsFutures.add(runAsync(() -> glueClient.updateColumnStatisticsForPartition(
+                        new UpdateColumnStatisticsForPartitionRequest()
+                                .withCatalogId(catalogId)
+                                .withDatabaseName(partition.getDatabaseName())
+                                .withTableName(partition.getTableName())
+                                .withPartitionValues(partition.getValues())
+                                .withColumnStatisticsList(columnChunk)),
+                        writeExecutor)));
 
-            Map<String, HiveColumnStatistics> currentColumnStatistics = this.getPartitionColumnStatisticsIfPresent(partition).orElse(ImmutableMap.of());
-            Set<String> removedStatistics = difference(currentColumnStatistics.keySet(), updatedColumnStatistics.keySet());
-            List<CompletableFuture<Void>> deleteStatsFutures = removedStatistics.stream()
-                    .map(column -> runAsync(() ->
-                            glueClient.deleteColumnStatisticsForPartition(new DeleteColumnStatisticsForPartitionRequest()
-                                    .withCatalogId(catalogId)
-                                    .withDatabaseName(partition.getDatabaseName())
-                                    .withTableName(partition.getTableName())
-                                    .withPartitionValues(partition.getValues())
-                                    .withColumnName(column)), writeExecutor))
-                    .collect(toUnmodifiableList());
+        Map<String, HiveColumnStatistics> currentColumnStatistics = this.getPartitionColumnStatisticsIfPresent(partition).orElse(ImmutableMap.of());
+        Set<String> removedStatistics = difference(currentColumnStatistics.keySet(), columnStatistics.keySet());
+        removedStatistics.forEach(column ->
+                operationsFutures.add(runAsync(() -> glueClient.deleteColumnStatisticsForPartition(
+                        new DeleteColumnStatisticsForPartitionRequest()
+                                .withCatalogId(catalogId)
+                                .withDatabaseName(partition.getDatabaseName())
+                                .withTableName(partition.getTableName())
+                                .withPartitionValues(partition.getValues())
+                                .withColumnName(column)),
+                        writeExecutor)));
+        getFutureValue(allOf(operationsFutures.toArray(CompletableFuture[]::new)));
```

Comment:
This will propagate failures as RuntimeExceptions - should be proper TrinoExceptions IMO.

Reply:
Good catch. 
Above we wrap whole call with 
```java
        catch (RuntimeException ex) {
            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
        }
```